src/consensus/tx_verify.cpp

```diff
@@ -351,14 +375,59 @@ bool Consensus::CheckTxInputs(const CTransaction& tx, CValidationState& state, c
             return state.DoS(100, false, REJECT_INVALID, "bad-txns-inputvalues-outofrange");
         }
 
+        if(coin.out.scriptPubKey.IsAssetScript()){
```

Comment:
move this to different function as well, and it seem that this a repeated pattern so i think better to share the same function. i understand it has different error message but the pattern almost identical

Reply:
done. moved this repeated code to a single function to handle inputs/outputs checks